### ISSUE TITLE
potential optimization of 'L'

### ISSUE THREAD

**Issue by Clayblockunova** [fasttime/JScrewIt#28]
afaik, call Object.prototype.toString on object Location (which equals to window.location) in Edge will return "[object Location]", and Function("return ({}).toString.call(location)") only cost 5066 char in non-IE browser, and currently 'L' cost 6181 char in non-IE browser when web worker supported. since Location is web-worker-safe,  definition of 'L' can be optimized with Location. But, Firefox and Safari not tested, so idk how practical it is.

**Comment by fasttime** [comment]
`Object.prototype.toString.call(location)` is "[object Location]" in the main thread of browsers other than IE, and "[object WorkerLocation]" in web workers. I added two [new definitions](https://github.com/fasttime/JScrewIt/blob/f40046bf4aac1b169a2560adbb7f208c1486b649/src/lib/definitions.js#L841-L844) that work with both variants.

**Comment by Clayblockunova** [comment]
why do you add definitions based on `String.prototype.at`? Definitions of "L" based on `Object.prototype.toString.call(location)` is supposed to work in Safari before 10 where  `atob()` only available outside web workers, and `String.prototype.at` is still unsupported by current browsers.

**Comment by fasttime** [comment]
Combinations of features that are not available in known environments are still interesting for a few reasons. Some features may be polyfilled in old browsers (e.g. by core-js). Some combinations may still be available in browsers that are not officially supported. In the end, the API allows using arbitrary features at the same time, as long as they don't mutually exclude each other. For these cases, definitions are not limited to known environments.

**Comment by fasttime** [comment]
Implemented in JScrewIt 2.28.

**Comment by Clayblockunova** [comment]
Sorry to bother you again.  afaik, `location.constructor` is `Location()` (a standard constructor Function) In Edge(Chromium), and `location.constructor.name` only cost 2180 chars non-IE browser. But, Firefox and Safari's case still unknown, and idk whether constructor function based method would make `Object.prototype.toString.call()` based method useless.

**Comment by fasttime** [comment]
I checked the value of `location.constructor + ''` in all browsers and this is what I got:

Browser | `location.constructor + ''`
-|-
Internet Explorer | `[object Location]`
Safari < 10 (main thread) | `[object LocationConstructor]`
Safari < 10 (web workers) | `[object WorkerLocationConstructor]`
other browsers | `function Location()` + ...

I haven't checked the encoding lengths yet, but all of these should be useful for new specific definitions.
On the other hand, I don't see how those could make any of the existing definitions useless. Specifically, I'm thinking that the "L" in `location.constructor + ''` could be located at index 8, 9 or 14 depending on the environment (and counting backwards gives even more possibilities...). Although it would be possible to use environment-sensitive paddings to align the "L" at the same position, the result would be probably too expensive compared to the current definition `Function("return toString.call(location)")()[SLICE_OR_SUBSTR]("-10")[1]`, which works in all browsers except IE.
